presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ApplyConnectorOptimization.java

```diff
@@ -292,4 +299,26 @@ private static <T> boolean containsAll(Set<T> container, Collection<T> test)
         }
         return true;
     }
+
+    private static class ConnectorToInternalJoinRewriter
+            extends SimplePlanRewriter<Void>
+    {
+        @Override
+        public PlanNode visitConnectorJoinNode(ConnectorJoinNode node, RewriteContext<Void> context)
+        {
+            return new JoinNode(node.getSourceLocation(),
+                    node.getId(),
+                    node.getStatsEquivalentPlanNode(),
+                    node.getType(),
+                    context.rewrite(node.getSources().get(0)),
+                    context.rewrite(node.getSources().get(1)),
+                    ImmutableList.copyOf(node.getCriteria()),
+                    node.getOutputVariables(),
+                    node.getFilters().stream().reduce(LogicalRowExpressions::and),
```

Comment:
nit: There is a variant which has a `Collection` as its input, so `Optional.of(LogicalRowExpressions.and(node.getFilters()))` will suffice

Reply:
It ends up being a little more code since I need to check if the filters list is empty. 
```
node.getFilters().isEmpty() ? Optional.empty() : Optional.of(LogicalRowExpressions.and(node.getFilters())),
``` 
Which do you prefer?

Reply:
I went with the second option so that it will create a flat Expression instead of a nested one.